src/coreclr/gc/gcpriv.h

```diff
@@ -171,10 +171,8 @@ inline void FATAL_GC_ERROR()
 // turned on.
 #define FEATURE_LOH_COMPACTION
 
-#ifdef FEATURE_64BIT_ALIGNMENT
 // We need the following feature as part of keeping 64-bit types aligned in the GC heap.
 #define RESPECT_LARGE_ALIGNMENT //Preserve double alignment of objects during relocation
```

Comment:
I do not think we want to be enabling this for x86. It would be an unnecessary perf regression on x86. The alignment is only a hint on x86, not a hard requirement.

Reply:
Is the regression due to the cost of ensuring 8-byte alignment in memory or the potential increase in memory usage introduced by the alignment padding?

Reply:
Both. This feature makes the GC heap compaction conservative. It both takes more time and more memory.

Reply:
Yeah, I was uncertain of the performance implications but I wanted to check if previously not having this was related to the GC assert I was seeing in the tests. @Maoni0 had the same impression as you that we should not be enabling it.

Given that all current usage of ALLOC_ALIGN8 guarantees preserving 8 byte alignment during relocation it feels odd to me that omitting the RESPECT_LARGE_ALIGNMENT we'd now be giving ALLOC_ALIGN8 different meanings depending on which architecture the GC is running on. It makes me question if we should be making this change at all vs. just leaving the existing workaround in place.

Reply:
> we'd now be giving ALLOC_ALIGN8 different meanings depending on which architecture the GC is running on

I do not think it is a problem. The clarifying comment you have added makes it clear.